### PR TITLE
rozanagy-FTS-21

### PR DESCRIPTION
**Pull request by rozanagy**
style(levelup):format Badge, BadgeLevel, Person, PersonDTO  …
refactor(levelup):rename Level class to BadgeLevel
feat(levelup):create PersonDTO
feat(levelup):create badgeLibrary.html
feat(levelup):update BadgeLibraryController

### CODE REVIEW

File: src/main/java/com/greenfoxacademy/levelup/controller/BadgeLibraryController.java [hunk 1]
```diff
@@ -0,0 +1,32 @@
+package com.greenfoxacademy.levelup.controller;
+
+
```
Comment (hulfyback) [inline src/main/java/com/greenfoxacademy/levelup/controller/BadgeLibraryController.java:3]: empty line... 😆

File: src/main/java/com/greenfoxacademy/levelup/controller/BadgeLibraryController.java [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+import com.greenfoxacademy.levelup.model.Badge;
+import com.greenfoxacademy.levelup.model.BadgeLevel;
+import org.springframework.stereotype.Controller;
+import org.springframework.ui.Model;
+import org.springframework.web.bind.annotation.GetMapping;
+
+import java.util.ArrayList;
+import java.util.List;
+
+
```
Comment (hulfyback) [inline src/main/java/com/greenfoxacademy/levelup/controller/BadgeLibraryController.java:13]: EL

File: src/main/java/com/greenfoxacademy/levelup/controller/BadgeLibraryController.java [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+    Badge badge1 = new Badge("1.0", "Super Badge", "god", levels);
+    Badge badge2 = new Badge("2.0", "Best Mentor", "mentor", levels);
+    Badge badge3 = new Badge("3.0", "Best Colleague", "general", levels);
+    Badge badge4 = new Badge("4.0", "Lame Jokes", "general", levels);
+    List<Badge> badges = new ArrayList<>();
+    badges.add(badge1);
+    badges.add(badge2);
+    badges.add(badge3);
+    badges.add(badge4);
+    model.addAttribute("badges", badges);
```
Comment (hulfyback) [inline src/main/java/com/greenfoxacademy/levelup/controller/BadgeLibraryController.java:29]: why didn't you outsource the logic into a Utility class? 🤔 just like in the case of rest controllers

File: src/main/java/com/greenfoxacademy/levelup/model/Badge.java [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+  }
+
+  public Badge(String version, String name, String tag, List<BadgeLevel> levels) {
+    this.version = version;
+    this.name = name;
+    this.tag = tag;
+    this.levels = levels;
+  }
+
+  public Badge() {
```
Comment (hulfyback) [inline src/main/java/com/greenfoxacademy/levelup/model/Badge.java]: maybe this empty constructor should be in the first place... but I'm not sure.. 😄

File: src/main/java/com/greenfoxacademy/levelup/model/Badge.java [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+  }
+
+  public List<BadgeLevel> getLevels() {
+    return levels;
+  }
+
+  public void setLevels(List<BadgeLevel> levels) {
+    this.levels = levels;
+  }
+}
```
Comment (hulfyback) [inline src/main/java/com/greenfoxacademy/levelup/model/Badge.java:68]: EL @ EOF

File: src/main/java/com/greenfoxacademy/levelup/model/BadgeLevel.java [hunk 6]
```diff
@@ -0,0 +1,61 @@
...
+  }
+
+  public List<Person> getHolders() {
+    return holders;
+  }
+
+  public void setHolders(List<Person> holders) {
+    this.holders = holders;
+  }
+}
```
Comment (hulfyback) [inline src/main/java/com/greenfoxacademy/levelup/model/BadgeLevel.java:61]: EL @ EOF

File: src/main/java/com/greenfoxacademy/levelup/model/Person.java [hunk 7]
```diff
@@ -0,0 +1,61 @@
...
+  }
+
+  public List<Badge> getBadges() {
+    return badges;
+  }
+
+  public void setBadges(List<Badge> badges) {
+    this.badges = badges;
+  }
+}
```
Comment (hulfyback) [inline src/main/java/com/greenfoxacademy/levelup/model/Person.java:61]: EL @ EOF

File: src/main/java/com/greenfoxacademy/levelup/model/PersonDTO.java [hunk 8]
```diff
@@ -0,0 +1,30 @@
...
+  }
+
+  public String getName() {
+    return name;
+  }
+
+  public void setName(String name) {
+    this.name = name;
+  }
+}
```
Comment (hulfyback) [inline src/main/java/com/greenfoxacademy/levelup/model/PersonDTO.java:30]: EL @ EOF

**Review by hulfyback — COMMENTED** [review]
Seems almost completely **OK** for me 🙂 just **empty lines** as usual 😃 about the convention of the order of the default (empty) constructor and the other constructors actually I didn't find anything 🤷🏿‍♂️

**Review by bekobarna — APPROVED** [review]

File: src/main/java/com/greenfoxacademy/levelup/controller/BadgeLibraryController.java (hunk 3 above)
Comment (rozanagy) [reply src/main/java/com/greenfoxacademy/levelup/controller/BadgeLibraryController.java:29]: because....

**Pull request merged**